crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +157,505 @@ where
+
+    apply_op_back(
+        py,
+        new_dag,
+        &std_gate(py, StandardGate::CZGate)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn swap_replacement_dag(py: Python) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &std_gate(py, StandardGate::SwapGate)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rxx_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RXXGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn ryy_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RYYGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rzz_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RZZGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rzx_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q0])?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q1])?;
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RZXGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q0])?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q1])?;
+
+    Ok(new_dag.clone())
+}
+
+#[pymodule]
 pub fn gate_direction(m: &Bound<PyModule>) -> PyResult<()> {
     m.add_wrapped(wrap_pyfunction!(py_check_with_coupling_map))?;
     m.add_wrapped(wrap_pyfunction!(py_check_with_target))?;
```

Comment:
Are we using these two methods outside of Rust? I don't see them being used in the python `gate_direction.py` fie.

Reply:
Yes, these are used in `check_gate_direction.py`. Both `CheckGateDirection` and `GateDirection` are implemented in `gate_direction.rs`. The logic is very similar for both, so I figured it makes sense to put those together and they are rather small each.